migrations/20231130142117_responsibilties.js

```diff
@@ -0,0 +1,156 @@
+
+exports.up = async function (knex) {
+  await knex.schema.createTable('responsibilities', table => {
+    table.increments().primary()
+    table.text('title').notNullable()
+    table.text('description')
+    table.text('type')
+    table.timestamp('created_at')
+    table.timestamp('updated_at')
+    table.bigInteger('group_id').references('id').inTable('groups')
+  })
+
+  const responsibilitiesData = [
+    {
+      title: 'Administration',
+      description: 'Allows for editing group settings, exporting data, and deleting the group.',
+      type: 'system'
+    },
+    {
+      title: 'Add Members',
+      description: 'The ability to invite and add new people to the group, and to accept or reject join requests.',
+      type: 'system'
+    },
+    {
+      title: 'Remove Members',
+      description: 'The ability to remove a member from the group.',
+      type: 'system'
+    },
+    {
+      title: 'Manage Content',
+      description: 'Adjust group topics, custom views and manage content that contradicts the agreements of the group.',
+      type: 'system'
+    }
+  ]
+
+  // Insert responsibilities into the 'responsibilities' table
+  await knex('responsibilities').insert(responsibilitiesData)
+
+  await knex.schema.createTable('common_roles', table => {
+    table.increments().primary()
+    table.text('name').notNullable()
+    table.text('description')
+    table.text('emoji')
+    table.timestamp('created_at')
+    table.timestamp('updated_at')
+  })
+
+  const commonRolesData = [
+    {
+      name: 'Manager',
+      description: 'The manager is empowered to do everything related to group administration.',
+      emoji: '💼'
+    },
+    {
+      name: 'Moderator',
+      description: 'Moderators are expected to actively engage in discussion, encourage participation, and take corrective action if a member violates group agreements.',
+      emoji: '🎤'
+    },
+    {
+      name: 'Host',
+      description: 'Hosts are responsible for cultivating a good atmosphere by welcoming and orienting new members, embodying the group culture and agreements, and helping members connect with relevant content and people.',
+      emoji: '🎉'
+    }
+  ]
+
+  // Insert common roles into the 'common_roles' table
+  await knex('common_roles').insert(commonRolesData)
+
+  await knex.schema.createTable('group_roles_responsibilities', table => {
+    table.increments().primary()
+    table.bigInteger('group_role_id').references('id').inTable('groups_roles')
```

Comment:
so for every referenced column we for some reason need to add something like this to have deletes work for the test database:
  await knex.raw('alter table groups_roles alter constraint groups_roles_group_id_foreign deferrable initially deferred')